modules/internal/ChapelRange.chpl

```diff
@@ -573,7 +613,17 @@ module ChapelRange {
     if chpl__singleValIdxType(idxType) {
       if _low > _high { // avoid circularity of calling .size which calls .high
         warning("This range is empty and has a single-value idxType, so its high bound isn't trustworthy");
-        return chpl_intToIdx(_low);
+        return lowBound;
+      }
+    }
+    if (boundedType == BoundedRangeType.boundedLow ||
+        boundedType == BoundedRangeType.boundedNone) {
+      if isEnumType(idxType) {
+        return chpl_intToIdx(idxType.size - 1);
+      } else if isBoolType(idxType) {
+        return chpl_intToIdx(1);
+      } else {
+        compilerError("Querying the high bound of a range not thought to have one");
```

Comment:
Please consider making this message more formal, less colloquial.

Reply:
Done in 0a245b81df611da5202e59316d7cf61e36466fb0